Matpower export

rb004f opened the pull request:
@ccoffrin First cut at an matpower export. Take a look

ccoffrin commented on src/io/matpower.jl:
```diff
@@ -504,8 +506,8 @@ function mp2pm_dcline(data::Dict{String,Any})
-        delete!(dcline, "pmin")
-        delete!(dcline, "pmax")
+#        delete!(dcline, "pmin")
+#        delete!(dcline, "pmax")
```
I think leaving these in will risk confusion on which parameter PowerModels is actually using.  I recommend renaming to `mp_pmin` and `mp_pmax` to mitigate confusion.

ccoffrin commented on src/io/matpower.jl:
```diff
@@ -665,3 +667,419 @@ function merge_generic_data(data::Dict{String,Any})
 
+"Export a power flow case in matpower format"
+function export_matpower(io::IO, data::Dict{String,Any})
+
```
In stead of doing the mixed-unit conversion while writing the output, can you just call `make_mixed_units(data)` at this point?

ccoffrin commented on src/io/matpower.jl:
```diff
@@ -665,3 +667,419 @@ function merge_generic_data(data::Dict{String,Any})
+                    "\t", gen["gen_status"],
+                    "\t", (gen["pmax"]*mvabase),
+                    "\t", (gen["pmin"]*mvabase),
+                    "\t", gen["pc1"],
```
By the technical Matpower spec I believe this is where the `haskey` checks should start.

ccoffrin commented on src/io/matpower.jl:
```diff
@@ -665,3 +667,419 @@ function merge_generic_data(data::Dict{String,Any})
+                    "\t", branch["t_bus"], 
+                    "\t", branch["br_r"], 
+                    "\t", branch["br_x"], 
+                    "\t", (haskey(branch,"b_to") ? branch["b_to"] + branch["b_fr"]  : 0), 
```
I don't believe `b_to` is not optional, I think you can hard code this as `branch["b_to"] + branch["b_fr"]`

ccoffrin commented on src/io/matpower.jl:
```diff
@@ -665,3 +667,419 @@ function merge_generic_data(data::Dict{String,Any})
+            println(io, "%    2    startup    shutdown    n    c(n-1)    ...    c0")
+            println(io, "mpc.dclinecost = [")
+        
+            for (idx,dcline) in (sort(dclines))
```
This block will always be identical to the generator one, might want to make it a function.

ccoffrin commented on test/data/matpower/extra_components.m:
```diff
@@ -0,0 +1,67 @@
+%% MATPOWER Case Format : Version 2
+mpc.version = '2';
```
case3.m is nominally the place to test the extended Matpower files.  Can you extend case3.m to cover your cases instead of adding a new case?

ccoffrin commented on test/export.jl:
```diff
@@ -0,0 +1,97 @@
+    close(io)    
+    destination_data = PowerModels.parse_file(temp_file)  
+    
+    @test InfrastructureModels.compare_dict(source_data, destination_data) == true
```
I think you can omit the `== true`

ccoffrin commented on test/export.jl:
```diff
@@ -0,0 +1,97 @@
+
```
These tests would fit well in `test/data.jl`

rb004f commented:
@ccoffrin pull request now addresses your comments. I will note that by changing case3.m, I had to change two other tests.  You may want to double check them.
